File: C++/towerOfHanoi.cpp
#include <iostream>

using namespace std;

void TOH(int n, char a, char b, char c){
    if(n==1){
        cout<<"Move disk 1 from "<<a<<" to "<<c<<endl;
        return;
    }
    TOH(n-1,a,c,b);
    cout<<"Move disk "<<n<<" from "<<a<<" to "<<c<<endl;
    TOH(n-1,b,a,c);
}

int main()
{
    int n;
    cin>>n;
    TOH(n,'A','B','C');
    return 0;
}

File: PYTHON/RecursiveAlgorithms/TowerOfHanoi.py
def TowerOfHanoi(n,from_pol, to_pol ,aux_pol):
    if n==0:
        return 0
    TowerOfHanoi(n-1,from_pol,aux_pol,to_pol)
    print("Move disk",n,"from pole",from_pol,"to pole",to_pol)
    TowerOfHanoi(n-1,aux_pol,to_pol,from_pol)
    return

TowerOfHanoi(3,'A','B','C')